pytorch_lightning/loops/fit_loop.py

```diff
@@ -218,7 +218,8 @@ def reset(self) -> None:
     def on_run_start(self) -> None:  # type: ignore[override]
         """Calls the ``on_train_start`` hook."""
         # reset train dataloader and val dataloader
-        self.trainer.reset_train_val_dataloaders(self.trainer.lightning_module)
+        self.trainer.reset_train_dataloader(self.trainer.lightning_module)
+        self.epoch_loop.val_loop._reload_evaluation_dataloaders()
```

Comment:
Can we defer this call to the moment validation loop actually runs?

Reply:
sorry, what do you mean? I made this change for the case when `num_sanity_steps=0`, the validation dataloaders are loaded here, but we need to use the loop method to load them to ensure that the tracker is updated correctly.

Reply:
My question is, why call this here in `FitLoop.on_run_start` instead of on `FitLoop.EpochLoop.EvaluationLoop.on_run_start`?

Reply:
for the progress bar. The main progress bar requires the total batches for both which is created during training.